### PR TITLE
Set up CI/CD pipeline with GitHub Actions

### PR DESCRIPTION
**Pull request by omkardongre**
Create ci.yml for Continuous Integration
Configure separate jobs for backend and frontend 
 <div id='description'>
<h3>Summary by Bito</h3>
A new CI workflow has been introduced for both backend and frontend, running on Ubuntu with Node.js 14. It includes jobs for dependency installation, linting, and frontend build checks. The workflow triggers on pushes and pull requests to main and develop branches.
<br>
<br>
<b>Code change type</b>: Configuration Changes, CI/CD Implementation
<br>
<br>
<b>Unit tests added</b>: False
<br>
<br>
<b>Estimated effort to review (1-5, lower is better)</b>: 1
</div>

### CODE REVIEW

**Comment by omkardongre** [comment]
## Code Review Agent Run Status

- **Limitations and other issues:** <small>❌</small>&nbsp;Failure - The AI Code Review Agent skipped reviewing this change because it is configured to exclude certain pull requests based on the source/target branch or the pull request status. You can change the settings [here](https://alpha.bito.ai/home/ai-agents/configured-agents/1), or contact the agent instance creator at [tesla111598@gmail.com](mailto:tesla111598@gmail.com).

**Comment by omkardongre** [comment]
<div>
<h2>Code Review Agent Run #f974b3</h2>
<div>
<ul>

<li>
<b>AI Based Review: </b><small>✔️</small>&nbsp;Successful
</li>

</ul>
</div>
<div>
<h3>High-level Feedback</h3>
The addition of a CI workflow is a positive step towards ensuring code quality and automating the build process. However, there are two key areas for improvement. Firstly, incorporate unit tests for the backend to enhance code reliability. Secondly, update the Node.js version from 14 to a supported LTS version (e.g., 18 or 20) to address potential security vulnerabilities and ensure compatibility. These changes will significantly strengthen the CI pipeline and improve overall project robustness.
</div>
<br>

<div>
<h3>Actionable Issues</h3>

<div id="issue">
<div>
:page_facing_up: github/workflows/ci.yml
</div>
<details>
<summary>
Issues: Total - 2, High importance - 2
</summary>
<div>


<table width="100%">
<tbody>

<tr>
<td width="30%">
<a href ="https://github.com/omkardongre/Food-Ordering-App/pull/3/files#r1760651544">Line 22-22</a>
</td>
<td width="70%">
 <small>🔴</small> High importance - 1 &nbsp;&nbsp; 
</td>
</tr>

<tr>
<td width="30%">
<a href ="https://github.com/omkardongre/Food-Ordering-App/pull/3/files#r1760651558">Line 28-28</a>
</td>
<td width="70%">
 <small>🔴</small> High importance - 1 &nbsp;&nbsp; 
</td>
</tr>

</tbody>
</table>

</div>
</details>
</div>

</div>




<span><i>AI Code Review powered by</i> <sub><sub><a href="https://bito.ai/" target="_blank">Bito Logo</a></sub></sub></span>

</div>

File: github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      run:
+        working-directory: ./backend
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Use Node.js
+      uses: actions/setup-node@v2
+      with:
+        node-version: '14'
```
Comment (omkardongre) [inline github/workflows/ci.yml]: <div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/3#issuecomment-2352179665>#f974b3</a><span> - 09/16/2024, 07:22 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The specified Node.js version (14) is outdated and no longer supported, which may lead to security vulnerabilities and compatibility issues.
</div>
<div id="fix">
<b>Fix:</b> Update the Node.js version to a currently supported LTS version, such as 18.x or 20.x.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -19,7 +19,7 @@
      - name: Use Node.js
        uses: actions/setup-node@v2
        with:
 -        node-version: '14'
 +        node-version: '18'
 @@ -39,7 +39,7 @@
      - name: Use Node.js
        uses: actions/setup-node@v2
        with:
 -        node-version: '14'
 +        node-version: '18'
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged

File: github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    - name: Install dependencies
+      run: npm ci
+
+    - name: Linting
+      run: npm run lint
```
Comment (omkardongre) [inline github/workflows/ci.yml:28]: <div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/3#issuecomment-2352179665>#f974b3</a><span> - 09/16/2024, 07:22 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The CI pipeline lacks a step for running unit tests for the backend, which is crucial for ensuring code quality and catching bugs early.
</div>
<div id="fix">
<b>Fix:</b> Add a step to run unit tests for the backend after the linting step.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -28,6 +28,9 @@
      - name: Linting
        run: npm run lint
 
 +    - name: Run unit tests
 +      run: npm test
 +
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged

Comment (ghost) [reply github/workflows/ci.yml:28]: tests are not added yet

**Pull request merged**